onetimepass/otp.py

```diff
@@ -225,47 +252,100 @@ def import_(ctx: click.Context, file, format_: str):
     pass
 
 
+# TODO(khanek) Move to separate module
+def get_alias_from_uri(uri: str) -> AliasSchema:
+    uri_parsed = otp_auth_uri.parse(uri)
+    return create_alias_schema(
+        otp_type=uri_parsed.otp_type,
+        label=uri_parsed.label,
+        issuer=uri_parsed.issuer,
+        secret=uri_parsed.secret,
+        digits_count=uri_parsed.digits,
+        hash_algorithm=uri_parsed.algorithm,
+        params=uri_parsed.params,
+    )
+
+
+# TODO(khanek) Move to separate module
+def create_alias(
+    otp_type: OTPType, label: str, issuer: str, secret: str, params: OTPParams
+) -> AliasSchema:
+    return create_alias_schema(
+        otp_type=otp_type,
+        label=label,
+        issuer=issuer,
+        secret=secret,
+        digits_count=settings.DB_DEFAULT_DIGITS_COUNT,
+        hash_algorithm=settings.DB_HASH_ALGORITHM,
+        params=params,
+    )
+
+
 @otp.command(help="Add the new secret as the specified ALIAS.")
 @click.argument("alias")
 @click.option("uri", "-u", "--uri", is_flag=True)
 @click.option("label", "-l", "--label")
 @click.option("period", "-p", "--period", type=click.INT)
+@click.option("counter", "-c", "--counter", type=click.INT)
 @click.option("issuer", "-i", "--issuer")
+@click.option("otp_type", "-t", "--otp-type", type=click.Choice(OTPType))
 @click.pass_context
 def add(
-    ctx: click.Context, alias: str, uri: bool, label: str, period: int, issuer: str
+    ctx: click.Context,
+    alias: str,
+    uri: bool,
+    label: str,
+    period: int,
+    counter: int,
+    issuer: str,
+    otp_type: OTPType,
 ):
-    # TODO(khanek) POC
+    quiet = ctx.obj["quiet"]
+
     db = JSONEncryptedDB(path=settings.DB_PATH, key=keyring_get().encode())
     data = get_db_data(db)
-    data.add_totp_alias(
-        name=alias,
-        secret=secrets.token_hex(),
-        digits_count=6,
-        hash_algorithm="sha1",
-        initial_time=datetime.datetime.fromtimestamp(0, tz=datetime.timezone.utc),
-        time_step_seconds=30,
-    )
-    db.write(data)
-    return  # TODO(khanek) Remove after POC
-
-    quiet = ctx.obj["quiet"]
+    if alias in data.otp:
+        raise click.UsageError(f"Alias {alias} exists. Consider renaming it")
 
     if uri:
         handle_conflicting_options(
             {
                 "-u, --uri": uri,
                 "-l, --label": label,
                 "-p, --period": period,
+                "-c, --counter": counter,
                 "-i, --issuer": issuer,
+                "-t, --otp-type": otp_type,
             }
         )
         input_uri = click.prompt("Enter URI", confirmation_prompt=True, hide_input=True)
+        alias_data = get_alias_from_uri(input_uri)
     else:
         input_secret = click.prompt(
             "Enter secret", confirmation_prompt=True, hide_input=True
         )
+        params = (
+            {
+                "initial_time": datetime.datetime.fromtimestamp(
```

Comment:
Maybe let's extract this as the option w/ default? I think the user should be able to control this parameter, even if it's not supported by the Key URI schema.